kubeconfig-creator/README.md

```diff
@@ -0,0 +1,155 @@
+`kubeconfig file` for the cluster and the `test-task` uses that kubeconfig file and verifiy whether the
+application has an access to the cluster or not by using some `kubectl/oc` commands.
+
+```
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: kubeconfig-test-pipeline
+spec:
+  workspaces:
+    - name: shared-workspace
+  tasks:
+    - name: kubeconfig-creator
+      taskRef:
+        name: kubeconfig-creator
+      workspaces:
+        - name: kubeconfigFile
+          workspace: shared-workspace
+      params:
+        - name: name
+          value: cluster-bot
+        - name: username
+          value: kubeadmin
+        - name: url
+          value: https://api.ci-ln-r68sydb-d5d6b.origin-ci-int-aws.dev.rhcloud.com:6443
+        - name: cadata
+          value: LS0tLS1....
+        - name: token
+          value: Xj95RM...
+    - name: authentication-test
+      taskRef:
+        name: authentication-test
+      workspaces:
+        - name: kubeconfigFile
+          workspace: shared-workspace
+      params:
+        - name: filename
+          value: kubeconfig
+      runAfter:
+        - kubeconfig-creator
+```
+`Test-task` uses shared-workspace to fetch the kubeconfig file form the
+kubeconfigFile directory and uses oc commands to check whether the `cluster` is configured or not.
+```
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: authentication-test
+spec:
+  params:
+    - name: filename
+      description: kubeconfig file name
+      type: string
+  workspaces:
+    - name: kubeconfigFile
+      readOnly: true
+  steps:
+    - name: get
+      image: quay.io/openshift/origin-cli:latest
+      script: |
+
+        export KUBECONFIG="$(workspaces.kubeconfigFile.path)/$(inputs.params.filename)"
+        #
+        # check that the cluster is configured
+        oc get pods
+```
+
+Workspace with `PVC` is used, as shown below.
+```
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: kubeconfig-pvc
+spec:
+  resources:
+    requests:
+      storage: 5Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadWriteOnce
+   ```
+   
+ Finally, Pipelinerun is used to execute the tasks in the pipeline and get the results.
+ ```
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: kubeconfig-test-pipeline-run
+spec:
+  pipelineRef:
+    name: kubeconfig-test-pipeline
+  workspaces:
+    - name: shared-workspace
+      persistentvolumeclaim:
+        claimName: kubeconfig-pvc
+```
+***Note*** : This `task` uses [kubeconfigwriter](https://github.com/tektoncd/pipeline/blob/master/cmd/kubeconfigwriter/main.go) image from the tektoncd pipeline project, this [image](https://console.cloud.google.com/gcr/images/tekton-releases/GLOBAL/github.com/tektoncd/pipeline/cmd/kubeconfigwriter) will only be updated in next release of the pipeline, so changes in the tektoncd pipeline code might not reflect immediately in the image.
```

Comment:
I think we will hold this PR until 0.12 is released.  So I suggest removing this note.  This Task will only become available to catalog users once kubeconfigwriter is updated.  What do you think?

Reply:
Initially, I thought of creating this PR after the 0.12 release, but some previous functionality can like accessing the cluster with `tokens` can be used. So created this PR and planned to updated once we get the 0.12 release.

But as you suggested, will update all these once we get the release.